### DISCUSSION TITLE
RayDistributor for using Ray to distribute the calculations in tsfresh

### DISCUSSION THREAD

**Discussion by TheaperDeng** [blue-yonder/tsfresh#944] [Ideas]
Ray is getting popular for building distributed applications and easy to fit into tsfresh by a `RayDistributor`.
I wonder if it's possible to have a `RayDistributor` upstream to tsfresh, I am willing to raise a PR and complete the test and doc.

An first version implementation is here https://github.com/TheaperDeng/ray_tsfresh

# Distributed tsfresh on Ray
This repo involves a new `RayDistributor` for tsfresh to use ray to distribute the calculations.

`RayDistributor` is a subclass of `IterableDistributorBaseClass` in tsfresh which follows the developing instruction in https://tsfresh.readthedocs.io/en/latest/text/tsfresh_on_a_cluster.html.

## Quick Start
Use `RayDistributor` the same way as `MultiprocessingDistributor`, `ClusterDaskDistributor` or `LocalDaskDistributor`.
```python
from ray_tsfresh import RayDistributor

distributor = RayDistributor(n_workers=8)
# ...
extracted_features = extract_features(..., distributor=distributor)
# ...
```

**Comment by nils-braun** [comment]
Hi @TheaperDeng ! That is actually really really nice! I am very sorry I did never answer - I have just missed this post.
I like this - would you like to include this into tsfresh as a PR?

**Reply by TheaperDeng** [reply]
Sure I am happy to do that. I will prepare a PR later in a few days.

**Reply by TheaperDeng** [reply]
A PR is raised in https://github.com/blue-yonder/tsfresh/pull/1030
